test/FunctionalTests/Microsoft.OData.Core.Tests/Json/ODataJsonServiceDocumentSerializerTests.cs

```diff
@@ -25,7 +25,7 @@ public void FunctionImportShouldWriteOutNameAndUrlCorrectly()
                 {
                     FunctionImports = new ODataFunctionImportInfo[] {new ODataFunctionImportInfo() {Name = "name1", Url = new Uri("http://Service/FunctionImport")}}
                 },
-                @"﻿{""value"":[{""name"":""name1"",""kind"":""FunctionImport"",""url"":""http://service/FunctionImport""}]}");
+                @"{""value"":[{""name"":""name1"",""kind"":""FunctionImport"",""url"":""http://service/FunctionImport""}]}");

```

Comment:
I can't spot the changes in this file...

Reply:
Seems like the original version has an unprintable character between `"{`. Not sure how it got there or why it didn't throw an error.

Reply:
When I restore the original version, the test fails:

![image](https://github.com/OData/odata.net/assets/8460169/973288af-98b6-43a1-9e5d-f8d41595e2a1)

Reply:
The test passes when using `JsonWriter` so this is an interesting catch.

Reply:
That character is a [byte order mark](https://en.wikipedia.org/wiki/Byte_order_mark). And I just remembered that I already had a discussion with you about this.

Reply:
In this case, the BOM is inserted by the `StreamWriter`. The BOM is based on the "preamble" returned by the `Encoding` instance's `GetPreamble()` value. By default `StreamWriter` uses an instance of `UTF8Encoding` that does not have a BOM. But if you pass a different encoder to the constructor, it might result in the BOM being written. It so happens that the instance returned by `Encoding.UTF8` has a preamble and results in the BOM being inserted at the front of the stream. And that's why the original tests passed for `JsonWriter` which uses a `StreamWriter` and don't pass for `ODataUtf8JsonWriter` unless we remove the BOM.

The reason we didn't have to do this across all our tests is because a lot of our tests either don't specify a BOM or don't use the StreamWriter, e.g. the `JsonWriterTests` use a `StringWriter`.